src/cider/nrepl/middleware/log.clj

```diff
@@ -0,0 +1,293 @@
+                        (event/exception-frequencies))})
+
+(defn frameworks-reply
+  "Return the available log frameworks."
+  [{:keys [session]}]
+  {:log-frameworks (->> (meta session)
+                        ::frameworks vals
+                        (map select-framework))})
+
+(defn format-event-reply
+  "Format a log event."
+  [{:keys [::print/print-fn] :as msg}]
+  (let [event (event msg)
+        writer (StringWriter.)]
+    (print-fn event writer)
+    {:log-format-event (str writer)}))
+
+(defn inspect-event-reply
+  "Inspect a log event."
+  [msg]
+  (inspect-value msg (event msg)))
+
+(defn levels-reply
+  "Return the log levels and their frequencies for the given framework and appender."
+  [msg]
+  {:log-levels (->> (framework/events (framework msg) (appender msg))
+                    (event/level-frequencies))})
+
+(defn loggers-reply
+  "Return the loggers and their frequencies for the given framework and appender."
+  [msg]
+  {:log-loggers (->> (framework/events (framework msg) (appender msg))
+                     (event/logger-frequencies))})
+
+(defn remove-appender-reply
+  "Remove an appender from a log framework."
+  [msg]
+  (let [appender (appender msg)]
+    (swap-framework! msg framework/remove-appender appender)
+    {:log-remove-appender {:id (str (:id appender))}}))
+
+(defn remove-consumer-reply
+  "Remove a consumer from the appender of a log framework."
+  [msg]
+  (let [appender (appender msg)
+        consumer (consumer msg)]
+    (swap-framework! msg framework/remove-consumer appender consumer)
+    {:log-remove-consumer (select-consumer consumer)}))
+
+(defn update-appender-reply
+  "Update the appender of a log framework."
+  [msg]
+  (let [appender (appender msg)]
+    {:log-update-appender
+     (-> (swap-framework! msg framework/update-appender appender)
+         (framework/appender appender)
+         (deref)
+         (select-appender))}))
+
+(defn update-consumer-reply
+  "Update the consumer of a log appender."
+  [msg]
+  (let [appender (appender msg)
+        consumer (consumer msg)]
+    {:log-update-consumer
+     (-> (swap-framework! msg framework/update-consumer appender consumer)
+         (framework/consumer appender consumer)
+         (select-consumer))}))
+
+(defn search-reply
+  "Search the log events of an appender."
+  [msg]
+  {:log-search
+   (->> (select-keys msg [:filters :limit :offset])
+        (framework/search-events (framework msg) (appender msg))
+        (map select-event))})
+
+(defn threads-reply
+  "Return the threads and their frequencies for the given framework and appender."
+  [msg]
+  {:log-threads (->> (framework/events (framework msg) (appender msg))
+                     (event/thread-frequencies))})
+
+(defn handle-log
+  "Handle NREPL log operations."
+  [handler {:keys [session] :as msg}]
+  (when-not (contains? (meta session) ::frameworks)
+    (alter-meta! session assoc ::frameworks (framework/resolve-frameworks)))
+  (with-safe-transport handler msg
```

Comment:
I'd suggest prepending all op names with`cider/`, so they are properly namespaced. See #710 for details. We should gradually move in this direction.

Reply:
I now added the cider prefix to the ops and the response keys.